app/models/vote.rb

```diff
@@ -0,0 +1,6 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  belongs_to :work
+  validates :work, uniqueness: { scope: :user }
+  validates :user, uniqueness: { scope: :work }
```

Comment:
Good work getting this tricky `uniqueness` scope figured out! However, I think you only need one or the other of these lines, not both.